tck/src/test/java/org/eclipse/microprofile/jwt/tck/container/jaxrs/AudValidationBadAudTest.java

```diff
@@ -0,0 +1,127 @@
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.tck.container.jaxrs;
+
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.PrivateKey;
+import java.util.HashMap;
+import java.util.Properties;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.jwt.config.Names;
+import org.eclipse.microprofile.jwt.tck.TCKConstants;
+import org.eclipse.microprofile.jwt.tck.config.SimpleTokenUtils;
+import org.eclipse.microprofile.jwt.tck.util.MpJwtTestVersion;
+import org.eclipse.microprofile.jwt.tck.util.TokenUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.Assert;
+import org.testng.Reporter;
+import org.testng.annotations.Test;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.microprofile.jwt.tck.TCKConstants.TEST_GROUP_CONFIG;
+
+/**
+ * Validate the handling of the JWT aud claim.
+ *
+ * Validate the aud claim against the {@linkplain Names#AUDIENCES} property is performed, and fails
+ * if the jwt aud does not match the {@linkplain Names#AUDIENCES} property.
+ */
+public class AudValidationBadAudTest extends Arquillian {
+    /**
+     * The base URL for the container under test
+     */
+    @ArquillianResource
+    private URL baseURL;
+
+    /**
+     * The token used by the test
+     */
+    private static String token;
+
+    /**
+     * Create a CDI aware base web application archive that includes an embedded PEM public key
+     * that is included as the mp.jwt.verify.publickey property.
+     * The root url is /
+     * @return the base base web application archive
+     * @throws Exception - on resource failure
+     */
+    @Deployment()
+    public static WebArchive createDeployment() throws Exception {
+        URL publicKey = AudValidationBadAudTest.class.getResource("/publicKey4k.pem");
+
+        PrivateKey privateKey = TokenUtils.readPrivateKey("/privateKey4k.pem");
+        String kid = "publicKey4k";
+        HashMap<String, Long> timeClaims = new HashMap<>();
```

Comment:
This is a minor one, but I've been keeping updating other TCK source :-), so please have `Map<...>` here and other new tests

Reply:
done